### ISSUE TITLE
D9VK is confused by DSR and tries to stretch Fullscreen-Window to the nvidia super resolution when the desktop is already at its native res (Game in question is GW2 on Win7)

### ISSUE THREAD

**Issue by Squall-Leonhart** [Joshua-Ashton/d9vk#122]
When D9VK creates a borderless window framebuffer, it applies the highest resolution available instead of retrieving and applying the current desktop resolution leading to the image being stretched offscreen - this is a problem for users who have an amd or nvidia super resolution configured but not currently applied.
![unknown 1](https://user-images.githubusercontent.com/1098176/58065077-c7a99980-7bc7-11e9-87bd-f56b931cb1b6.png)

### Software information
Guild Wars 2

### System information
- GPU: GTX 1080
- Driver: r430.53 and .64
- Wine version: Not applicable
- DXVK version:  d9vk-master.bee2c9a3c1abeebf37fa16b5b6d29e7718d3a70b

### Log files
[Gw2-64_d3d9.log](https://github.com/Joshua-Ashton/d9vk/files/3200549/Gw2-64_d3d9.log)

**Comment by Squall-Leonhart** [comment]
As can be seen in the log, the wrapper is projecting to a 4k window when it should inherit the current desktop resolution

info:  D3D9: Setting display mode: 2560x1440@120
warn:  D3D9DeviceEx::SetMaterial: Stub
warn:  D3D9SwapChainEx::GetGammaRamp: Stub
warn:  D3D9SwapChainEx::GetGammaRamp: Stub
info:  Presenter: Actual swap chain properties:
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  2542x1361
  Image count:  3
info:  D3D9DeviceEx::ResetEx:
  Requested Presentation Parameters
    - Width:              3840
    - Height:             2160
    - Format:             X8R8G8B8
    - Auto Depth Stencil: true
    - Windowed:           true

warn:  D3D9DeviceEx::SetMaterial: Stub
warn:  D3D9SwapChainEx::GetGammaRamp: Stub
warn:  D3D9SwapChainEx::GetGammaRamp: Stub
info:  Presenter: Actual swap chain properties:
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  3840x1462
  Image count:  3

**Comment by Squall-Leonhart** [comment]
Without DSR, it retrieves the correct 1440p mode, since its the  highest avaiable.

warn:  D3D9DeviceEx::SetMaterial: Stub
warn:  D3D9SwapChainEx::GetGammaRamp: Stub
warn:  D3D9SwapChainEx::GetGammaRamp: Stub
info:  Presenter: Actual swap chain properties:
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  2560x1440
  Image count:  3
info:  D3D9DeviceEx::ResetEx:
  Requested Presentation Parameters
    - Width:              2542
    - Height:             1361
    - Format:             X8R8G8B8
    - Auto Depth Stencil: true
    - Windowed:           true

warn:  D3D9DeviceEx::SetMaterial: Stub
warn:  D3D9SwapChainEx::GetGammaRamp: Stub
warn:  D3D9SwapChainEx::GetGammaRamp: Stub
info:  Presenter: Actual swap chain properties:
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_FIFO_KHR
  Buffer size:  2542x1361
  Image count:  3

[Gw2-64_d3d9.log](https://github.com/Joshua-Ashton/d9vk/files/3200566/Gw2-64_d3d9.log)

**Comment by Joshua-Ashton** [comment]
Borderless windowed mode is nothing to do with me, it's literally an application thing.

If the game specifies however that it doesn't want me to change its window... Then maybe I can do something. Otherwise you're pooped.

**Comment by Joshua-Ashton** [comment]
Actually... another idea.

**Comment by Joshua-Ashton** [comment]
Can you try this commit? https://github.com/Joshua-Ashton/d9vk/commit/62ba8e94d3d6a4b322131de9570c534fce6b3195

On the `wip-dsr-fix` branch

**Comment by Squall-Leonhart** [comment]
hey, sure i can test it if it gets built on your froggi or I can wait on a friend to spin me up a  build.

I created an issue is because without d9vk, the game renders to 1440p as expected, so if its not your code its upstream?

**Comment by Joshua-Ashton** [comment]
its not built on froggie. I can build for you when i get home

**Comment by Squall-Leonhart** [comment]
thanks.

**Comment by DxCx** [comment]
Hey,

Using Wine-tkg 4.6 i see the same behavior (native resolution is 4k, but i work with 1080p, desktop and game)

so, after seen this thread i had to give it a try.
first of all, here is the compiled binary: 
[wip-dsr-fix-62ba8e94.zip](https://github.com/Joshua-Ashton/d9vk/files/3203143/wip-dsr-fix-62ba8e94.zip)

however, using the patch i can see the same behavior, also logs show that:
```
info:  Game: GW2-64.exe
info:  DXVK: v1.2.1
warn:  OpenVR: Failed to locate module
info:  Enabled instance extensions:
info:    VK_KHR_get_physical_device_properties2
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
info:  GeForce GTX 1050:
info:    Driver: 418.52.3
info:    Vulkan: 1.1.106
info:    Memory Heap[0]: 
info:      Size: 4096 MiB
info:      Flags: 0x1
info:      Memory Type[7]: Property Flags = 0x1
info:      Memory Type[8]: Property Flags = 0x1
info:    Memory Heap[1]: 
info:      Size: 11947 MiB
info:      Flags: 0x0
info:      Memory Type[0]: Property Flags = 0x0
info:      Memory Type[1]: Property Flags = 0x0
info:      Memory Type[2]: Property Flags = 0x0
info:      Memory Type[3]: Property Flags = 0x0
info:      Memory Type[4]: Property Flags = 0x0
info:      Memory Type[5]: Property Flags = 0x0
info:      Memory Type[6]: Property Flags = 0x0
info:      Memory Type[9]: Property Flags = 0x6
info:      Memory Type[10]: Property Flags = 0xe
info:  Process set as DPI aware
info:  Game: GW2-64.exe
info:  DXVK: v1.2.1
warn:  OpenVR: Failed to locate module
info:  Enabled instance extensions:
info:    VK_KHR_get_physical_device_properties2
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
info:  GeForce GTX 1050:
info:    Driver: 418.52.3
info:    Vulkan: 1.1.106
info:    Memory Heap[0]: 
info:      Size: 4096 MiB
info:      Flags: 0x1
info:      Memory Type[7]: Property Flags = 0x1
info:      Memory Type[8]: Property Flags = 0x1
info:    Memory Heap[1]: 
info:      Size: 11947 MiB
info:      Flags: 0x0
info:      Memory Type[0]: Property Flags = 0x0
info:      Memory Type[1]: Property Flags = 0x0
info:      Memory Type[2]: Property Flags = 0x0
info:      Memory Type[3]: Property Flags = 0x0
info:      Memory Type[4]: Property Flags = 0x0
info:      Memory Type[5]: Property Flags = 0x0
info:      Memory Type[6]: Property Flags = 0x0
info:      Memory Type[9]: Property Flags = 0x6
info:      Memory Type[10]: Property Flags = 0xe
info:  Process set as DPI aware
info:  Enabled device extensions:
info:    VK_EXT_conditional_rendering
info:    VK_EXT_depth_clip_enable
info:    VK_EXT_host_query_reset
info:    VK_EXT_shader_viewport_index_layer
info:    VK_EXT_transform_feedback
info:    VK_EXT_vertex_attribute_divisor
info:    VK_KHR_dedicated_allocation
info:    VK_KHR_descriptor_update_template
info:    VK_KHR_draw_indirect_count
info:    VK_KHR_driver_properties
info:    VK_KHR_get_memory_requirements2
info:    VK_KHR_image_format_list
info:    VK_KHR_maintenance1
info:    VK_KHR_maintenance2
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_shader_draw_parameters
info:    VK_KHR_swapchain
info:  DXVK: Read 4294 valid state cache entries
info:  DXVK: Using 4 compiler threads
warn:  D3D9: VK_FORMAT_D16_UNORM_S8_UINT -> VK_FORMAT_D32_SFLOAT_S8_UINT
info:  D3D9DeviceEx::ResetEx:
  Requested Presentation Parameters
    - Width:              3840
    - Height:             2160
    - Format:             X8R8G8B8
    - Auto Depth Stencil: true
    - Windowed:           false

warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 9
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 16
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 37
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 38
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 60
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 136
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 137
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 141
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 142
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 145
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 146
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 154
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 155
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 158
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 161
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 166
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 172
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 173
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 178
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 179
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state 182
warn:  D3D9DeviceEx::SetTextureStageState: Stub
info:  Presenter: Actual swap chain properties:
  Format:       VK_FORMAT_B8G8R8A8_UNORM
  Present mode: VK_PRESENT_MODE_IMMEDIATE_KHR
  Buffer size:  1920x1080
  Image count:  2
debug: Hud: Enabled fps
debug: Hud: Enabled frametimes
info:  D3D9: Setting display mode: 3840x2160@60
warn:  D3D9DeviceEx::SetMaterial: Stub
debug: Compiling shader FS_01189c8325a6a74488eda9a1b499ceb96ec45d1e
debug: Compiling shader VS_1b12db6deb1325403d4a9f11d2d96b6f46db65ba
debug: Compiling shader FS_b7f70bb47b22ec9d8f3672e68349513c40eed014
debug: Compiling shader VS_1a93a94038473b209d13a7242973e947e0ed4802
debug: Compiling shader FS_4e82d3a5834581818584e1415f935688c6ec1688
debug: Compiling shader FS_b6418497676a1d039ae11ae58d7745bb8b9ce08b
debug: Compiling shader VS_1e4a040d5c738199368c028761ddad36415dcfb4
debug: Compiling graphics pipeline...
```

Since its wine and not windows i think that its still worth @Squall-Leonhart to give it a try as well.

**Comment by Squall-Leonhart** [comment]
Stretching for me still too.

its fine in pure fullscreen, it's just Borderless receiving the highest possible available resolution as the requested presentation parameter instead of recieving what the desktop is configured to.

**Comment by DxCx** [comment]
So for me its not causing any stretched image (even before the patch)
Its just annoying to have change in resolution while in game, especially when i play on the same resoultion as desktop

**Comment by DxCx** [comment]
So for me its not causing any stretched image (even before the patch)
Its just annoying to have change in resolution while in game, especially when i play on the same resoultion as desktop

**Comment by Squall-Leonhart** [comment]
[Gw2-64_d3d9.log](https://github.com/Joshua-Ashton/d9vk/files/3203278/Gw2-64_d3d9.log)

**Comment by Squall-Leonhart** [comment]
ok, so the new log doesn't appear any different to the old one

[Gw2-64_d3d9.log](https://github.com/Joshua-Ashton/d9vk/files/3203291/Gw2-64_d3d9.log)

**Comment by DxCx** [comment]
Btw on standard wine implemention (just removing d9vk dll)
The resolution change doesnt happen

**Comment by DxCx** [comment]
@Joshua-Ashton not sure which patch fixed that since there are couple of commits regarding resize, but using latest master (b1ce57c42cde01d8c99a14cf7cac5e1da4819d5d) the issue is fixed for me, no more resize =)

Updated log:
[GW2-64_d3d9.log](https://github.com/Joshua-Ashton/d9vk/files/3209296/GW2-64_d3d9.log)

@Squall-Leonhart can you also give it a try i guess :)

**Comment by Squall-Leonhart** [comment]
its gone in the opposite direction for me, the image is even more stretched now and i can't even click on the options to flip it back to full screen now, log output has changed though.

[Gw2-64_d3d9.log](https://github.com/Joshua-Ashton/d9vk/files/3209336/Gw2-64_d3d9.log)

**Comment by aqxa1** [comment]
A likely related issue: Ducktales: Remastered is a game that renders at 1080p only, but can scale to your desktop resolution. With wined3d this scaling works correctly, but with D9VK, the game is rendered in a box in the top left corner, and also looks to be stretched/cuttoff.

![Ducktales 1440p output](https://i.imgur.com/dHBqFnb.jpg)

If you select 1080p it's also rendered in a box at the top left, but there is no cutoff output. On the other hand, there is garbage in the rest of the screen.

![Ducktales 1080p output](https://i.imgur.com/Jfogsy3.jpg)

I also tested forcing virtual desktop mode (didn't help) but haven't tested vanilla Wine as of yet, only Proton.

System:
Proton 4.11
D9VK master
Mesa ACO git (also affects LLVM)
Vega 56

**Comment by Squall-Leonhart** [comment]
Funfact, this has actually been fixed since the merge into dxvk.
